vpr/src/route/connection_router.cpp

```diff
@@ -413,8 +444,33 @@ void ConnectionRouter<Heap>::timing_driven_expand_neighbours(t_heap* current,
                                        cost_params,
                                        bounding_box,
                                        target_node,
-                                       target_bb);
+                                       target_bb,
+                                       route_tree_nodes);
+    }
+}
+
+template<typename Heap>
+bool ConnectionRouter<Heap>::node_exists_in_route_tree(t_heap* current,
```

Comment:
This is the type of method that belong in a dedicate path data management object, rather than in ConnectionRouter